run_tests.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+import argparse
+import contextlib
+import io
+import logging
+import sys
+from pathlib import Path
+
+from src.main import main as decompile
+from src.options import Options
+
+CRASH_STRING = "CRASHED\n"
+
+
+def set_up_logging(debug: bool) -> None:
+    logging.basicConfig(
+        format="[%(levelname)s] %(message)s",
+        level=logging.DEBUG if debug else logging.INFO,
+    )
+
+
+def decompile_and_compare(
+    asm_file_path: Path, output_path: Path, should_overwrite: bool
+) -> bool:
+    logging.debug(
+        f"Decompiling {asm_file_path}"
+        + (f" into {output_path}" if should_overwrite else "")
+    )
+    try:
+        original_contents = output_path.read_text()
+    except FileNotFoundError:
+        logging.info(f"{output_path} does not exist. Creating...")
+        original_contents = "(file did not exist)"
+
+    final_contents = decompile_and_capture_output(output_path, asm_file_path)
+
+    if should_overwrite:
+        output_path.write_text(final_contents)
+
+    changed = final_contents != original_contents
+    if changed:
+        logging.info(
+            "\n".join(
+                [
+                    f"Output of {asm_file_path} changed!",
+                    "PREVIOUS:",
+                    original_contents,
+                    "NEW:",
+                    final_contents,
+                ]
+            )
+        )
+    return changed
+
+
+def decompile_and_capture_output(output_path: Path, asm_file_path: Path) -> str:
+    out_string = io.StringIO()
+    with contextlib.redirect_stdout(out_string):

```

Comment:
redirect stderr too?

Reply:
Our code-output pipeline only goes to stdout at the moment, right? Redirecting stderr would be good for debugging crashes, though, yeah.